spec/market_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'pry'
+require 'rspec'
+require './lib/item'
+require './lib/vendor'
+require './lib/market'
+
+RSpec.describe Market do
+  before(:each) do
+    @item1 = Item.new({ name: 'Peach', price: '$0.75' })
+    @item2 = Item.new({ name: 'Tomato', price: '$0.50' })
+    @item3 = Item.new({ name: 'Peach-Raspberry Nice Cream', price: '$5.30' })
+    @item4 = Item.new({ name: 'Banana Nice Cream', price: '$4.25' })
+    @vendor1 = Vendor.new('Rocky Mountain Fresh')
+    @vendor2 = Vendor.new('Ba-Nom-a-Nom')
+    @vendor3 = Vendor.new('Palisade Peach Shack')
+    @market = Market.new('South Pearl Street Farmers Market')
+  end
+
+  it 'is a Market' do
+    expect(@market).to be_an_instance_of(Market)
+    expect(@market.name).to eq('South Pearl Street Farmers Market')
+  end
+  it 'can add vendors' do
+    @vendor1.stock(@item1, 35)
+    @vendor1.stock(@item2, 7)
+    @vendor2.stock(@item4, 50)
+    @vendor2.stock(@item3, 25)
+    @vendor3.stock(@item1, 65)
+    @market.add_vendor(@vendor1)
+    @market.add_vendor(@vendor2)
+    @market.add_vendor(@vendor3)
+    expect(@market.vendors.count).to eq(3)
+  end
+
+  it 'has vendors' do
+    expect(@market.vendor_names).to eq([])
+    @market.add_vendor(@vendor1)
+    @market.add_vendor(@vendor2)
+    @market.add_vendor(@vendor3)
+    expect(@market.vendor_names.length).to eq(3)
+    # binding.pry
+  end
+
+  it 'returns vendors that sell item' do
+    @vendor1.stock(@item1, 35)
+    @vendor1.stock(@item2, 7)
+    @vendor2.stock(@item4, 50)
+    @vendor2.stock(@item3, 25)
+    @vendor3.stock(@item1, 65)
+    @market.add_vendor(@vendor1)
+    @market.add_vendor(@vendor2)
+    @market.add_vendor(@vendor3)
+    expect(@market.vendors_that_sell(@item1)).to eq([@vendor1, @vendor3])
+    expect(@market.vendors_that_sell(@item4)).to eq([@vendor2])
+  end
+
+  it 'can sort items' do
+    @vendor1.stock(@item1, 35)
+    @vendor1.stock(@item2, 7)
+    @vendor2.stock(@item4, 50)
+    @vendor2.stock(@item3, 25)
+    @vendor3.stock(@item1, 65)
+    @market.add_vendor(@vendor1)
+    @market.add_vendor(@vendor2)
+    @market.add_vendor(@vendor3)
+    expect(@market.sorted_items_list).to eq(['Banana Nice Cream', 'Peach', 'Peach-Raspberry Nice Cream', 'Tomato'])
+  end
+
+  it 'can check for overstock' do
+    @vendor1.stock(@item1, 35)
+    @vendor1.stock(@item2, 7)
+    @vendor2.stock(@item4, 50)
+    @vendor2.stock(@item3, 25)
+    @vendor3.stock(@item1, 65)
+    @market.add_vendor(@vendor1)
+    @market.add_vendor(@vendor2)
+    @market.add_vendor(@vendor3)
+    expect(@market.overstock_items).to eq([])
```

Comment:
The spec asks for #overstocked_items and #sorted_item_list. Try to make you method names match exactly!! On a real team, a developer working on an adjacent feature might have to change their method names if they dont match what is exactly specified.